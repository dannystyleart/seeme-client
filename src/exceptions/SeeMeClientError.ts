export class SeeMeClientError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'SeeMeClientError';
  }
}
