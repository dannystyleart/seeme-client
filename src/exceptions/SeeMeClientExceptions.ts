export enum SeeMeClientExceptions {
  API_KEY_MISSING = 'API_KEY_MISSING',
  API_HOST_MISSING = 'API_HOST_MISSING',
  API_HOST_PROTOCOL_UNSUPPORTED = 'API_HOST_PROTOCOL_UNSUPPORTED',
  API_PATH_INVALID = 'API_PATH_INVALID',
  SETIP_INVALID_IP = 'SETIP_INVALID_IP'
}
