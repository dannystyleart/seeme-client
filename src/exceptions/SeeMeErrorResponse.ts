export class SeeMeErrorResponse extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'SeeMeErrorResponse';
  }
}
