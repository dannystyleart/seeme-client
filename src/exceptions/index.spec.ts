import { SeeMeClientError } from './SeeMeClientError';
import { SeeMeErrorResponse } from './SeeMeErrorResponse';
import { SeeMeClientExceptions } from './SeeMeClientExceptions';
import * as exportedModules from './index';

describe('exceptions', () => {
  test('should export correct submodules', () => {
    expect(exportedModules).toMatchSnapshot();
  });

  test('SeeMeClientExceptions should contain correct enum values', () => {
    expect(SeeMeClientExceptions).toMatchSnapshot();
  });

  [SeeMeClientError, SeeMeErrorResponse].forEach((exceptionClass) =>
    test(`${exceptionClass.name} should be customized Error class`, () => {
      const instance = new exceptionClass('some error');
      expect(instance).toBeInstanceOf(Error);
      expect(instance).toMatchSnapshot();
    })
  );
});
