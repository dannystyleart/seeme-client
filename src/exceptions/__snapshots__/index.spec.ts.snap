// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exceptions SeeMeClientError should be customized Error class 1`] = `[SeeMeClientError: some error]`;

exports[`exceptions SeeMeClientExceptions should contain correct enum values 1`] = `
Object {
  "API_HOST_MISSING": "API_HOST_MISSING",
  "API_HOST_PROTOCOL_UNSUPPORTED": "API_HOST_PROTOCOL_UNSUPPORTED",
  "API_KEY_MISSING": "API_KEY_MISSING",
  "API_PATH_INVALID": "API_PATH_INVALID",
}
`;

exports[`exceptions SeeMeErrorResponse should be customized Error class 1`] = `[SeeMeErrorResponse: some error]`;

exports[`exceptions should export correct submodules 1`] = `
Object {
  "SeeMeClientError": [Function],
  "SeeMeClientExceptions": Object {
    "API_HOST_MISSING": "API_HOST_MISSING",
    "API_HOST_PROTOCOL_UNSUPPORTED": "API_HOST_PROTOCOL_UNSUPPORTED",
    "API_KEY_MISSING": "API_KEY_MISSING",
    "API_PATH_INVALID": "API_PATH_INVALID",
  },
  "SeeMeErrorResponse": [Function],
}
`;
